platform_services/test/test.js

```diff
@@ -13,13 +13,200 @@
+    var cbCheck = cbResponse.includes(errType) && cbResponse.includes(errMessage);
+    AWS.restore("DynamoDB.DocumentClient");
+    logStub.restore();
+    stub.restore();
+    assert.isTrue(logCheck && cbCheck);
+  });
+
+  /*
+  * Given an event.method = "GET" and invalid service_id, handler() attempts to get all items that match
+  * @param {object} event -> event.method is defined to be "GET", event.path.id is undefined
+  * @params {object, function} default aws context, and callback function as defined in beforeEach
+  */
+  it("should attempt to get all/filtered items from dynamoDB if 'GET' method and no id are defined", function(){
+    event.method = "GET";
+    event.path.id = undefined;
+    //mocking DynamoDB's scan function and wrapping with predefined spy
+    var mockDynamo = AWS.mock("DynamoDB", "scan", spy);
+    //trigger the spy by calling index.handler() with desired params
+    var callFunction = index.handler(event, context, callback);
+    AWS.restore("DynamoDB");
+    assert.isTrue(spy.called);
+  });
 
-describe('Sample', function() {
-    it('tests handler', function(done) {
-        // this.timeout(1000);
-        done();
+  /*
+  * Given a failed attempt at fetching data from DynamoDB, handler() should inform of error
+  * @param {object} event -> event.method is defined to be "GET", event.path.id is truthy
+  * @params {object, function} default aws context, and callback function as defined in beforeEach
+  * @returns {string} should return the callback response which is an error message
+  */
+  it("should indicate an InternalServerError occured if DynamoDB.scan fails", ()=>{
+    //creating an object with the callback function in order to wrap and test callback parameters
+    var callbackObj = {
+      "callback" : callback
+    };
+    event.method = "GET";
+    event.path.id = undefined;
+    errType = "InternalServerError";
+    errMessage = "unexpected error occured";
+    logMessage = "Error occured. ";
+    var err = {
+      "errorType" : "svtfoe",
+      "message" : "starco"
+    };
+    //mocking DynamoDB.scan, expecting callback to be returned with params (error,data)
+    AWS.mock("DynamoDB", "scan", (params, cb) => {
+      return cb(err);
     });
+    //wrapping the logger and callback function to check for response messages
+    stub = sinon.stub(callbackObj,"callback",spy);
+    logStub = sinon.stub(logger, "error", spy);
+    //trigger the mocked logic by calling handler()
+    var callFunction = index.handler(event, context, callbackObj.callback);
+    var logResponse = logStub.args[0][0];
+    var cbResponse = stub.args[0][0];
+    var logCheck = logResponse.includes(logMessage) && logResponse.includes(err.errorType) &&
+                    logResponse.includes(err.message);
+    var cbCheck = cbResponse.includes(errType) && cbResponse.includes(errMessage);
+    AWS.restore("DynamoDB");
+    logStub.restore();
+    stub.restore();
+    assert.isTrue(logCheck && cbCheck);
+  });
```

Comment:
This is a great start. Can you add a unit test case where "admin_users" can query all services? Similarly the second case where the dyanamodb table has records from multiple users and the GET only gets the specified users records.